### DISCUSSION TITLE
There is four questions I want to know!

### DISCUSSION THREAD

**Discussion by kstcmcjw** [nrc-cnrc/EGSnrc#842] [Questions]
Hello everyone, through those days study of EGS, i like this software step by step, programmed by myself, simulation the examples from getting-start.pdf, sometimes i succeeded, but there still many troubles i must to face. Some problems i ask below:

1. I simulation the chapter 5.4 (Generate particles tracks) of  getting-started.pdf , add the single geometry to the compound geometry one by one. But the EGS_VIEW faild, it was loading the .egsinp file a second and went to crash!
Here is my program:
    ```
    :start geometry definitaion:
    
        :start geometry:
            library = egs_box
            box size = 500 500 500
            minpoint=0 0 0 
            name = air_box
            :start media input:
                media = AIR521ICRU
            :stop media input:
        :stop geometry:
        
        :start geometry:
            library = egs_genvelope
            name = phantom_in_air
            base geometry = air_box
            inscribed geometries = phantom
        :stop geometry:

        simulation geometry=phantom_in_air

    :stop geometry definition:
    ```

    I think my geometry is right, but what is the mistake? And why the egs_view is crash either?

2. Is the library and type is the fixed format? For example, in EGS_ConeStack Class Reference, “library = egs_cones & type = EGS_ConeStack” ,it mustn’t be “library = egs_cones & type = EGS_Cones”?

3. Why the “axis” have six numbers, I mean axis = Ox, Oy, Oz,  ax, ay, az, I know the axis key defines the axis of all cones as a position (Ox,Oy,Oz) and direction (ax,ay,az), O stands for position, such as -1,2,0 in three-dimensional, but direction (ax,ay,az) stand for what? Such as “axis = 1.2417 0 0 -1 0 0”, what about “-1” means? Negative half shaft of X?

4. In EGS_PlanesT< T > Class Template Reference, the key value pairs is:
library = egs_planes
type = EGS_Xplanes or EGS_Yplanes or EGS_Zplanes or EGS_Planes
positions = list of plane co-ordinates
normal    = ax, ay, az (only needed for type = EGS_Planes)

    It seems there the plane’s position contains three site: ax, ay, az. But in some examples of Report PIRS-898 (2019) there is two positions as the follows:

    ![image](https://user-images.githubusercontent.com/96934217/152647621-daca0c52-d7f0-4388-b64b-e00428b7a36d.png)

Thanks for your answer!

**Comment by ftessier** [comment]
1. There is a typo in `:start geometry definitaion:` (definiti**~~a~~**on), I don't know if this will resolve the problem...
2. Yes, it is ConeStack for a cone stack; there are various geometries in the cones libary, the type allows you to pick which one.
3. The axis defines the axis _and_ direction of the set of cones; for example, `-1 0 0` means that the layers of a cone stack are added one after the other along towards the _negative_ x axis direction.
4. There can be 2 or more positions: these are the _distances_ along the axis (defined by the normal) where the planes are located: there are _n-1_ regiones defined by _n_ positions.

**Reply by kstcmcjw** [reply]
> 1. 有一个错字`:start geometry definitaion:`（定义**~一个~**on)，我不知道这是否能解决问题...
> 2. 是的，它是锥形堆的 ConeStack；圆锥库中有各种几何形状，类型允许您选择哪一种。
> 3. 轴定义了一组锥体的轴_和_方向；例如，`-1 0 0`表示锥形堆叠的层沿着_负_x轴方向一个接一个地添加。
> 4. 可以有 2 个或更多位置：这些是平面所在的沿轴（由法线定义）的_距离_：有_n-1 个_区域由_n 个_位置定义。

1. Oh, thank you very much, apologizes for my carelessness of spell mistake, first problem has solved, the program runs OK.
2. I search the library=egs_cones, find there are some types, such as type=EGS_conestock; type=EGS_simplecones and type=EGS_coneset,
there is another question, it seems the library=cylinders and library=egs_cones can all achieve the cylinder's simulation. But what is the different? Only is the simulation's process?

3.4. Yes, thank you for your advice, i simulation the geometry in my brain at first, and then draw a picture of 3 and 4 question.
![image](https://user-images.githubusercontent.com/96934217/153098315-723db988-7f20-40e5-8f13-9cd8d528d383.png)

5. There is two additional question i see the example of PIRS-898 (2019), what is this four numbers(0,0,1,0) stand for? The picture is below. 

6. For ax,ay,az or ox,oy,oz? And whether the ox,oy,oz is only two conditions-- -1 and 1, stand for the negtive and positive of axis?

![image](https://user-images.githubusercontent.com/96934217/153099039-a88498b7-f115-4ba3-b348-bf1b588c20a4.png)

**Reply by kstcmcjw** [reply]
> 1. 有一个错字`:start geometry definitaion:`（定义**~一个~**on)，我不知道这是否能解决问题...
> 2. 是的，它是锥形堆的 ConeStack；圆锥库中有各种几何形状，类型允许您选择哪一种。
> 3. 轴定义了一组锥体的轴_和_方向；例如，`-1 0 0`表示锥形堆叠的层沿着_负_x轴方向一个接一个地添加。
> 4. 可以有 2 个或更多位置：这些是平面所在的沿轴（由法线定义）的_距离_：有_n-1 个_区域由_n 个_位置定义。

Hello?
